test/e2e/operator/nmstate_install_test.go

```diff
@@ -130,6 +131,54 @@ var _ = Describe("NMState operator", func() {
 			})
 		})
 	})
+	Context("when cluser-reader exists", func() {
+		const (
+			clusterReaderRoleName = "cluster-reader"
+			testUserNamespace     = "default"
+			serviceAccountName    = "test-user"
+			testUserBindingName   = "test-user-binding"
+		)
+
+		var clusterReaderCreated bool
+
+		BeforeEach(func() {
+			err := createClusterReaderCR(clusterReaderRoleName)
+			Expect(err).To(SatisfyAny(Succeed(), WithTransform(apierrors.IsAlreadyExists, BeTrue())))
+			if err == nil {
+				clusterReaderCreated = true
+			}
+
+			Expect(createTestUserSA(testUserNamespace, serviceAccountName)).To(Succeed(),
+				"should success creating a serviceaccount")
+			Expect(createTestUserCRB(testUserBindingName, testUserNamespace, serviceAccountName, clusterReaderRoleName)).To(Succeed(),
+				"should success creating a clusterrolebinding")
+
+			By("Install NMState for the first time")
+			installNMState(defaultOperator.nmstate)
+			eventuallyOperandIsReady(defaultOperator)
+		})
+		AfterEach(func() {
+			uninstallNMStateAndWaitForDeletion(defaultOperator)
+		})
+		AfterEach(func() {
+			Expect(deleteTestUserCRB(testUserBindingName)).To(Succeed())
+		})
+		AfterEach(func() {
+			Expect(deleteTestUserSA(testUserNamespace, serviceAccountName)).To(Succeed())
+		})
+		AfterEach(func() {
+			if clusterReaderCreated {
+				Expect(deleteClusterReaderCR(clusterReaderRoleName)).To(Succeed())
+			}
+		})
```

Comment:
why Three `AfterEach` ?

Reply:
so if one fails, it will still try to clean the others